File: mobile-client/ios/UseCase/Auth/Session/StartAuthUseCase.swift
public struct StartAuthUseCase {
    private let gateway: AuthSessionGateway

    public init(gateway: AuthSessionGateway) {
        self.gateway = gateway
    }

    public func callAsFunction(request: StartAuthRequest) async throws -> AuthSessionPayload {
        try await gateway.startAuth(request: request)
    }
}

File: mobile-client/android/UseCase/Auth/Session/StartAuthUseCase.kt
package app.mobiling.client.usecase.auth.session

import app.mobiling.client.contract.auth.session.AuthSessionPayload
import app.mobiling.client.contract.auth.session.StartAuthRequest
import app.mobiling.client.data.auth.session.AuthSessionGateway

class StartAuthUseCase(
    private val gateway: AuthSessionGateway,
) {
    suspend operator fun invoke(request: StartAuthRequest): AuthSessionPayload = gateway.startAuth(request)
}
